### PR TITLE
Code version is not initialized in a pristine database.

### PR DESCRIPTION
**Pull request by azdagron**
The code version is not set in the migration table when the database is newly initialized. If the server is shut down and a version with an older schema is started, it will fail to start even if it is within the +/- minor version compatibility boundary, since the "code version" stored in the database will be effectively 0.0.0.

This change fixes the pristine database initialization code to initialize the code version column appropriately.

### CODE REVIEW

**Review by evan2645 — APPROVED** [review]
🥩 🦉 

thanks @azdagron!

File: pkg/server/plugin/datastore/sql/migration.go [hunk 1]
```diff
@@ -185,7 +185,10 @@ func initDB(db *gorm.DB, dbType string, log hclog.Logger) (err error) {
 		return sqlError.Wrap(err)
 	}
 
-	if err := tx.Assign(Migration{Version: latestSchemaVersion}).FirstOrCreate(&Migration{}).Error; err != nil {
+	if err := tx.Assign(Migration{
+		Version:     latestSchemaVersion,
+		CodeVersion: codeVersion.String(),
```
Comment (amartinezfayo) [inline pkg/server/plugin/datastore/sql/migration.go:190]: It would be great if this could be captured by a test.

Comment (azdagron) [reply pkg/server/plugin/datastore/sql/migration.go:190]: Yeah, i started down that road and didn't have a clear way to test for that other than reaching into the database manually. I'll add one though.

**Review by amartinezfayo — APPROVED** [review]
Thanks @azdagron!

**Pull request merged**